File: app/src/main/java/com/example/dagger_hilt/ui/activity/main/viewModel/MainViewModel.java
package com.example.dagger_hilt.ui.activity.main.viewModel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.example.dagger_hilt.data.datasource.database.entities.MovieEntity;
import com.example.dagger_hilt.domain.MovieRepository;
import com.example.dagger_hilt.sys.util.Resource;

import java.util.List;
import java.util.concurrent.Executors;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;

@HiltViewModel
public class MainViewModel extends ViewModel {
    private final MovieRepository repository;
    private final LiveData<Resource<List<MovieEntity>>> moviesList;

    @Inject
    public MainViewModel(MovieRepository repository) {
        this.repository = repository;
        moviesList = repository.loadMovies();
    }

    public void updateMovie(int id, boolean check) {
        var executor = Executors.newSingleThreadExecutor();
        executor.execute(() -> repository.updateMovie(id, check));
    }

    public LiveData<Resource<List<MovieEntity>>> getMoviesList() {
        return moviesList;
    }
}

File: app/src/main/java/com/example/gob_fact/ui/activity/main/MainViewModel.kt
package com.example.gob_fact.ui.activity.main

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.gob_fact.data.datasource.database.entities.FactEntity
import com.example.gob_fact.data.datasource.database.entities.MovieEntity
import com.example.gob_fact.domain.IFactRepository
import com.example.gob_fact.sys.util.Resource
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class MainViewModel @Inject constructor(
    factRepository: IFactRepository
) : ViewModel() {

    private val _facts: LiveData<List<FactEntity>> = factRepository.getFacts()
    val facts: LiveData<List<FactEntity>> get() = _facts

}
